src/vs/workbench/contrib/terminalContrib/suggest/browser/taskCompletionProvider.ts

```diff
@@ -0,0 +1,185 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CancellationToken } from '../../../../../base/common/cancellation.js';
+import { ITerminalCompletionProvider } from './terminalCompletionService.js';
+import { ITerminalCompletion, TerminalCompletionItemKind } from './terminalCompletionItem.js';
+import { ITaskService } from '../../../tasks/common/taskService.js';
+import { Task } from '../../../tasks/common/tasks.js';
+import { ICommandService } from '../../../../../platform/commands/common/commands.js';
+import { TerminalShellType } from '../../../../../platform/terminal/common/terminal.js';
+import { IThemeService } from '../../../../../platform/theme/common/themeService.js';
+import { ThemeIcon } from '../../../../../base/common/themables.js';
+import { Codicon } from '../../../../../base/common/codicons.js';
+
+export class TaskCompletionProvider implements ITerminalCompletionProvider {
```

Comment:
Could we move all the task completion provider into a new `terminalContrib/task/` folder? Make sure the new terminal contrib is included in `src\vs\workbench\contrib\terminal\terminal.all.ts`. Follow these principles:

- `terminalContrib/suggest/` should know about command-based completions
- `terminalContrib/task/` should know about tasks and provide command-based completions to `terminalContrib/suggest/`